File: apps/backend/scripts/init-db.py
#!/usr/bin/env python3
"""
Database initialization script using Python + SQLite3
Creates PathLife database with all required tables
"""

import sqlite3
import os
import sys

def init_database():
    # Paths
    script_dir = os.path.dirname(os.path.abspath(__file__))
    backend_dir = os.path.dirname(script_dir)
    db_path = os.path.join(backend_dir, 'pathlife.db')
    sql_path = os.path.join(backend_dir, 'init-db.sql')

    print('PathLife Database Initialization')
    print('=' * 50)
    print(f'\nDatabase path: {db_path}')
    print(f'SQL script: {sql_path}\n')

    # Check if SQL file exists
    if not os.path.exists(sql_path):
        print(f'Error: {sql_path} not found!')
        sys.exit(1)

    # Read SQL script
    with open(sql_path, 'r') as f:
        sql_script = f.read()

    # Create/connect to database
    print('Creating database...')
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()

    try:
        # Execute SQL script
        print('Executing SQL schema...')
        cursor.executescript(sql_script)
        conn.commit()
        print('✓ Database schema created successfully!')

        # Verify tables
        cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
        tables = cursor.fetchall()
        print(f'\n✓ Created {len(tables)} tables:')
        for table in tables:
            print(f'  - {table[0]}')

        print(f'\n✓ Database initialized at: {db_path}')
        print('\nYou can now start the application!')

    except sqlite3.Error as e:
        print(f'\n✗ Error creating database: {e}')
        sys.exit(1)

    finally:
        conn.close()

if __name__ == '__main__':
    init_database()

File: apps/backend/scripts/init-db.js
#!/usr/bin/env node

/**
 * Database initialization script
 * Creates SQLite database with all required tables
 */

const fs = require('fs');
const path = require('path');

// For environments where Prisma engines can't be downloaded
// We'll create a minimal SQLite database manually

const dbPath = path.join(__dirname, '..', 'pathlife.db');
const sqlPath = path.join(__dirname, '..', 'init-db.sql');

console.log('PathLife Database Initialization');
console.log('=================================\n');

console.log(`Database path: ${dbPath}`);
console.log(`SQL script: ${sqlPath}\n`);

// Check if SQL file exists
if (!fs.existsSync(sqlPath)) {
  console.error('Error: init-db.sql not found!');
  process.exit(1);
}

// Create empty database file
console.log('Creating database file...');
if (!fs.existsSync(dbPath)) {
  fs.writeFileSync(dbPath, '');
  console.log('✓ Database file created');
} else {
  console.log('✓ Database file already exists');
}

console.log('\n' + '='.repeat(50));
console.log('IMPORTANT: Database schema initialization');
console.log('='.repeat(50));
console.log('\nYou need to run the SQL schema manually:');
console.log('\n1. Install sqlite3:');
console.log('   brew install sqlite3     (macOS)');
console.log('   apt-get install sqlite3  (Linux)');
console.log('\n2. Initialize the schema:');
console.log(`   sqlite3 ${dbPath} < ${sqlPath}`);
console.log('\n3. Or use Prisma (if engines work):');
console.log('   npx prisma db push\n');
console.log('='.repeat(50) + '\n');
